Client/Frontend/Browser/URIFixup.swift

```diff
@@ -50,4 +62,18 @@ class URIFixup {
         }
         return components?.url
     }
+    
+    /// Checks whether a string is a valid email conforming to RFC 2822.
+    /// http://www.cocoawithlove.com/2009/06/verifying-that-string-is-email-address.html
+    static func isValidEmail(_ string: String) -> Bool {
+        let regex =
+            "(?:[a-zA-Z0-9!#$%\\&‘*+/=?\\^_`{|}~-]+(?:\\.[a-zA-Z0-9!#$%\\&'*+/=?\\^_`{|}" +
+            "~-]+)*|\"(?:[\\x01-\\x08\\x0b\\x0c\\x0e-\\x1f\\x21\\x23-\\x5b\\x5d-\\" +
+            "x7f]|\\\\[\\x01-\\x09\\x0b\\x0c\\x0e-\\x7f])*\")@(?:(?:[a-z0-9](?:[a-" +
+            "z0-9-]*[a-z0-9])?\\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\\[(?:(?:25[0-5" +
+            "]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-" +
+            "9][0-9]?|[a-z0-9-]*[a-z0-9]:(?:[\\x01-\\x08\\x0b\\x0c\\x0e-\\x1f\\x21" +
+            "-\\x5a\\x53-\\x7f]|\\\\[\\x01-\\x09\\x0b\\x0c\\x0e-\\x7f])+)\\])"
+        return NSPredicate(format: "SELF MATCHES[c] %@", regex).evaluate(with: string)
```

Comment:
Interesting, never thought of using `NSPredicate` here instead of `NSRegularExpression`, is there any diff?

Reply:
Hmm I never thought to use any of the other options.. however, it seems that `range(of:)` is actually the most optimized between all three with -O.. It doesn't let me specify -O3 but I'm sure with O3 it'd be negligible or produce the same results.

https://godbolt.org/z/p8XcFE

I'll switch to `range(of:)` which is more legible anyway.

Reply:
Fixed.